lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,455 @@
+            value_text_left_pos_x <- value_text_left_pos_x.map(|(left,dot)| -*left - *dot / 2.0);
+            eval value_text_left_pos_x((x) model.value_text_left.set_position_x(*x));
+            eval model.value_text_left.height((h) model.value_text_left.set_position_y(*h / 2.0));
+            eval model.value_text_dot.width((w) model.value_text_dot.set_position_x(-*w / 2.0));
+            eval model.value_text_dot.height((h) model.value_text_dot.set_position_y(*h / 2.0));
+            eval model.value_text_dot.width((w) model.value_text_right.set_position_x(*w / 2.0));
+            eval model.value_text_right.height((h) model.value_text_right.set_position_y(*h / 2.0));
+
+            model.label.set_content <+ input.set_label;
+            eval input.set_label_hidden((v) model.set_label_hidden(*v));
+            eval model.label.height((h) model.label.set_position_y(*h / 2.0));
+            label_pos_x <- all4(
+                &input.set_width,
+                &input.set_height,
+                &model.label.width,
+                &input.set_label_position,
+            ).map(
+                |(comp_width,comp_height,lab_width,position)| match *position {
+                    LabelPosition::Inside => -comp_width / 2.0 + comp_height / 2.0,
+                    LabelPosition::Outside => -comp_width / 2.0 - comp_height / 2.0 - lab_width,
+                }
+            );
+            eval label_pos_x((x) model.label.set_position_x(*x));
+        };
+        self.init_precision_defaults()
+    }
+
+    /// Initialize the precision adjustment areas above/below the slider and the default precision
+    /// value
+    fn init_precision_defaults(self) -> Self {
+        self.frp.set_default_precision(PRECISION_DEFAULT);
+        self.frp.set_precision_adjustment_margin(PRECISION_ADJUSTMENT_MARGIN);
+        self.frp.set_precision_adjustment_step_size(PRECISION_ADJUSTMENT_STEP_SIZE);
+        self
+    }
+}
+
+impl display::Object for Slider {
+    fn display_object(&self) -> &display::object::Instance {
+        self.model.display_object()
+    }
+}
+
+impl FrpNetworkProvider for Slider {
+    fn network(&self) -> &enso_frp::Network {
+        self.frp.network()
+    }
+}
+
+impl application::View for Slider {
+    fn label() -> &'static str {
+        "Slider"
+    }
+
+    fn new(app: &Application) -> Self {
+        Self::new(app)
+    }
+
+    fn app(&self) -> &Application {
+        &self.app
+    }
+}
+
+/// Position of the slider label.
+#[derive(Clone, Copy, Debug, Default)]
+pub enum LabelPosition {
+    #[default]
+    /// Place the label outside the slider component, on its left side.
+    Outside,
+    /// Place the label inside the slider component, on the left side.
+    Inside,
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl_core::define_endpoints_2! {
+    Input {
+        /// Set the width of the slider component.
+        set_width(f32),
```

Comment:
Minor comment - could you move the FPR definition above the slider please? This way, when reading the file you can first learn about the API and then see its usage. Its similar to a comment I did in the last review - to make definitions placed first before they are used, to allow the reader read the file top-bottom :)